tuf/api/metadata.py

```diff
@@ -250,59 +250,61 @@ def sign(
 
         return signature
 
-    def verify(
+    def verify_delegate(
         self,
-        key: Mapping[str, Any],
+        role_name: str,
+        delegate: "Metadata",
         signed_serializer: Optional[SignedSerializer] = None,
-    ) -> bool:
-        """Verifies 'signatures' over 'signed' that match the passed key by id.
+    ):
+        """Verifies that 'delegate' is signed with the required threshold of
+        keys for the delegated role 'role_name'.
 
-        Arguments:
-            key: A securesystemslib-style public key object.
-            signed_serializer: A SignedSerializer subclass instance that
-                implements the desired canonicalization format. Per default a
-                CanonicalJSONSerializer is used.
+        Args:
+            role_name: Name of the delegated role to verify
+            delegate: The Metadata object for the delegated role
+            signed_serializer: Optional; serializer used for delegate
+                serialization. Default is CanonicalJSONSerializer.
 
         Raises:
-            # TODO: Revise exception taxonomy
-            tuf.exceptions.Error: None or multiple signatures found for key.
-            securesystemslib.exceptions.FormatError: Key argument is malformed.
-            tuf.api.serialization.SerializationError:
-                'signed' cannot be serialized.
-            securesystemslib.exceptions.CryptoError, \
-                    securesystemslib.exceptions.UnsupportedAlgorithmError:
-                Signing errors.
-
-        Returns:
-            A boolean indicating if the signature is valid for the passed key.
-
+            UnsignedMetadataError: 'delegate' was not signed with required
+                threshold of keys for 'role_name'
         """
-        signatures_for_keyid = list(
-            filter(lambda sig: sig.keyid == key["keyid"], self.signatures)
-        )
-
-        if not signatures_for_keyid:
-            raise exceptions.Error(f"no signature for key {key['keyid']}.")
 
-        if len(signatures_for_keyid) > 1:
-            raise exceptions.Error(
-                f"{len(signatures_for_keyid)} signatures for key "
-                f"{key['keyid']}, not sure which one to verify."
+        # Find the keys and role in our metadata
+        role = None
+        if isinstance(self.signed, Root):
+            keys = self.signed.keys
+            role = self.signed.roles.get(role_name, None)
+        elif isinstance(self.signed, Targets):
+            if self.signed.delegations:
+                keys = self.signed.delegations.keys
+                # Assume role names are unique in delegations.roles: #1426
+                roles = self.signed.delegations.roles
+                role = next((r for r in roles if r.name == role_name), None)
+        else:
+            raise ValueError("Call is valid only on delegator metadata")
```

Comment:
from Trishank: "Why ValueError?"

Valid question as arguments are not the issue here (unless we consider "self" and argument of course). 

In any case it's not terribly important as this is a programming error and should not be handled by anyone

Reply:
I left it a ValueError: I'm not going to create a custom error for something that isn't meant to be caught and can't find a better one in the builtin exceptions.

Reply:
Could be `TypeError` since what `isinstance` checks is the type but is it much better than `ValueError` .. I guess not.

Reply:
I originally did not pick that because the object type the call is made on (Metadata) is correct ... but I think you're right, it's not perfect but it's better.